src/lib/server/preprocessMessages.ts

```diff
@@ -11,8 +11,14 @@ export async function preprocessMessages(
 ): Promise<Message[]> {
 	return await Promise.all(
 		structuredClone(messages).map(async (message, idx) => {
+			const webSearchContext = webSearch?.contextSources
+				.map(({ context }) => context)
+				.flat()
+				.sort((a, b) => a.idx - b.idx)
+				.map(({ text }) => text)
+				.join(" ");
```

Comment:
Is this the best place to do this, does it ensure that the right sources are mapped to the right context? 👀